tests/testsuite/build_script.rs

```diff
@@ -4464,67 +4502,6 @@ fn optional_build_script_dep() {
     p.cargo("run --features bar").with_stdout("1\n").run();
 }
 
-#[cargo_test]
-fn optional_build_dep_and_required_normal_dep() {
```

Comment:
I'm not clear why this test is being removed. It seems like it was added for a specific issue when mixing an optional with a non-optional test (#5487). Can you say why it is being removed instead of just updating the test?

Reply:
I guess it was my idea <https://github.com/rust-lang/cargo/pull/11252#issuecomment-1325705539>. Can bring them back if needed.

Reply:
I've brought the test back, and updated it to show that it doesn't get the build sharing optimization in this situation since there is no shared dependency in the first command.